schemas/iso19139/src/main/plugin/iso19139/layout/layout.xsl

```diff
@@ -631,4 +632,64 @@
                 priority="2100"/>
 
 
+  <!-- In Simple view, if INSPIRE setting is enabled use gn-checkbox-with-nilreason for gmd:pass (conformity) -->
+  <xsl:template mode="mode-iso19139"
+                match="gmd:pass[$tab='default' and /root/gui/settings/system/inspire/enable = 'true']"
+                priority="5000">
+
+    <xsl:variable name="xpath" select="gn-fn-metadata:getXPath(.)"/>
+    <xsl:variable name="isoType" select="if (@gco:isoType) then @gco:isoType else ''"/>
+    <xsl:variable name="labelConfig" select="gn-fn-metadata:getLabel($schema, name(), $labels, name(..), $isoType, $xpath)"/>
+
+    <xsl:variable name="nilReason" select="@gco:nilReason" />
+    <xsl:variable name="tagLabels">{"true": "conformant", "false": "notConformant", "unknown": "notEvaluated"}</xsl:variable>
+
+    <xsl:variable name="parentEditInfo" select="gn:element" />
+    <xsl:variable name="editInfo" select="gco:Boolean/gn:element" />
+
+    <xsl:variable name="elementId" select="concat('_X', $parentEditInfo/@ref, '_replace')" />
+
+    <xsl:variable name="isRequired" as="xs:boolean">
+      <xsl:choose>
+        <xsl:when
+          test="($parentEditInfo and $parentEditInfo/@min = 1 and $parentEditInfo/@max = 1) or
+          (not($parentEditInfo) and $editInfo and $editInfo/@min = 1 and $editInfo/@max = 1)">
+          <xsl:value-of select="true()"/>
+        </xsl:when>
+        <xsl:when test="gn-fn-metadata:getLabel($schema, name(), $labels, name(..),$isoType, $xpath)/condition = 'mandatory'">
+          <xsl:value-of select="true()"/>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:value-of select="false()"/>
+        </xsl:otherwise>
+      </xsl:choose>
+    </xsl:variable>
+
+
+    <xsl:variable name="directiveAttributes">
+      <directiveAttributes>
+        <attr name="label" value="{$labelConfig/label}" />
```

Comment:
Does it means that we have to change all config-editor directive configuration which is using the following structure:
```
<directiveAttributes
              data-template-add-action="true"
              data-search-action="false"
              data-popup-action="true"
              data-template-type="contact"
              data-filter='{"_root": "gmd:CI_ResponsibleParty"}'
              data-variables="gmd:role/gmd:CI_RoleCode/@codeListValue~{role}"
              data-transformation="contact-from-iso19139-to-iso19115-3"/>
```
?

Reply:
No, afaik the `render-element` with the parameter `directive` doesn't use that stuff from `config-editor.xml`, only adds the directive name: https://github.com/geonetwork/core-geonetwork/blob/3.4.x/web/src/main/webapp/xslt/ui-metadata/form-builder.xsl#L147

The `directiveAttributes` from `config-editor.xml` is used in the following code:

* `render-element-template-field`: https://github.com/geonetwork/core-geonetwork/blob/3.4.x/web/src/main/webapp/xslt/ui-metadata/form-configurator.xsl#L707-L715

* `render-element-to-add`: https://github.com/geonetwork/core-geonetwork/blob/3.4.x/web/src/main/webapp/xslt/ui-metadata/form-builder.xsl#L915-L922

Maybe can be changed to use the same stuff?

Reply:
Would make sense to have the same type of config for the same things.

Reply:
The current checkbox has some challenges, this PR seems to solve some items, so I'd be glad to see it improved. 
- the true/false value is currently not translated, adding a label for each state could be an option
- some users don't like the current single checkbox option, they prefer radio boxes with explicit options for true and false (and empty). 

Alternative option for empty/nill could be to add ctrl-click event to uncheck one of the radio's.